arrow-csv/src/writer.rs

```
@@ -400,30 +272,6 @@ impl WriterBuilder {
         self
     }
 
-    /// Set the CSV file's date format
```

Presumably since someone added these functions previously they were being used 🤔  On the other hand, given they aren't passed to the actual writer and the code was marked as dead maybe they were silently ignored